claims in-source build

There's an (annoying) block against in-source builds, that has false positives:
```
[~/c/rpma](master)$ cmake .
-- The C compiler identification is GNU 10.2.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
CMake Error at CMakeLists.txt:12 (message):
  In-source build is not allowed.


-- Configuring incomplete, errors occurred!
See also "/home/kilobyte/c/rpma/CMakeFiles/CMakeOutput.log".
[~/c/rpma](master)$ cd /tmp/
[/tmp]$ mkdir r
[/tmp]$ cd r
[/tmp/r]$ cmake ~/c/rpma/
CMake Error at CMakeLists.txt:12 (message):
  In-source build is not allowed.


-- Configuring incomplete, errors occurred!
See also "/home/kilobyte/c/rpma/CMakeFiles/CMakeOutput.log".
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,11 +7,6 @@ cmake_minimum_required(VERSION 3.3)
 
 project(rpma C)
 
-# disable in-source builds
-if(${PROJECT_SOURCE_DIR} STREQUAL ${PROJECT_BINARY_DIR})
-	message(FATAL_ERROR "In-source build is not allowed.")
-endif()
-
 set(VERSION_MAJOR 0)
 set(VERSION_MINOR 9)
 set(VERSION_PATCH 0)

